app/mailers/request_mailer.rb

```diff
@@ -231,10 +231,10 @@ def self.receive(raw_email, source = :mailin)
 
   # Find which info requests the email is for
   def requests_matching_email(email)
-    # We deliberately don't use Envelope-to here, so ones that are BCC
-    # drop into the holding pen for checking.
-    addresses = ((email.to || []) + (email.cc || [])).compact
-    InfoRequest.matching_incoming_email(addresses)
+    addresses = MailHandler.get_all_addresses(email)
+    addresses.concat (MailHandler.get_header_string('Received', email) || '').
+      scan(/[A-Z0-9+_.-]+@[A-Z0-9.-]+/i)
```

Comment:
I think there is an issue here when an email contains more than one `Received` header. `MailHandler.get_header_string('Received', email)` will return an array of Ruby objects which have been converted as string using `Array#to_s` EG. `[#<Mail::Field ... >", #<Mail::Field ... >"]`. We can't be sure the email address with be matched as this output could be truncated so need to handle multiple headers correctly as this will be the norm.

Internally for email regexp matching we normally use used `MySociety::Validate.email_find_regexp` which would be preferable. 

As it stands its not very clear looking at the code what it is doing, extracting to a helper method with a descriptive name would help but also worth considering the location of the code, I would probably move to be within `MailHandler` and refactor slightly to extract to a `#emails_within_received_headers` method. I think we could call `#email['received']` rather than calling `#get_header_string(...)` as we're doing with `envelope-to`, not sure if those are different or not..?